### PR TITLE
[automation] Added ItemStateUpdate action

### LINKED ISSUE #1926: Allow item update in UI rules
My suggestion is to include an "item update" action (that would execute a postUpdate command) in the following screen
imagem

### PR DESCRIPTION
**Pull request by cweitkamp**
- Added ItemStateUpdate action

Closes #1926

Signed-off-by: Christoph Weitkamp <github@christophweitkamp.de>

### CODE REVIEW

File: bundles/org.openhab.core.automation/src/main/java/org/openhab/core/automation/internal/module/handler/ItemUpdateActionHandler.java
Comment (kaikreuzer) [inline bundles/org.openhab.core.automation/src/main/java/org/openhab/core/automation/internal/module/handler/ItemUpdateActionHandler.java:34]:
```suggestion
 * This is an implementation of an ActionHandler. It sends state updates for items.
```

File: bundles/org.openhab.core.automation/src/main/java/org/openhab/core/automation/internal/module/handler/ItemUpdateActionHandler.java
Comment (kaikreuzer) [inline bundles/org.openhab.core.automation/src/main/java/org/openhab/core/automation/internal/module/handler/ItemUpdateActionHandler.java:41]: We have ItemStateEvents for this and we generally refer to ItemStateUpdates in order to distinguish them from ItemUpdates (i.e. changes to the item itself). I'd hence prefer to also keep it here this way and name this `core.ItemStateUpdateAction`.

File: bundles/org.openhab.core.automation/src/main/java/org/openhab/core/automation/internal/module/handler/ItemUpdateActionHandler.java
Comment (kaikreuzer) [inline bundles/org.openhab.core.automation/src/main/java/org/openhab/core/automation/internal/module/handler/ItemUpdateActionHandler.java:68]: The `ItemCommandActionHandler` does this check only within `execute`, but not within the constructor. I think that's the much better approach, because the item might not be defined at rule definition time, while this should not cause the rule to not initialize.

File: bundles/org.openhab.core.automation/src/main/java/org/openhab/core/automation/internal/module/handler/ItemUpdateActionHandler.java
Comment (kaikreuzer) [inline bundles/org.openhab.core.automation/src/main/java/org/openhab/core/automation/internal/module/handler/ItemUpdateActionHandler.java:49]: The item should not be cached here as it can potentially change over time. Get the latest item in `execute` instead.

File: bundles/org.openhab.core.automation/src/main/resources/OH-INF/automation/moduletypes/ItemActions.json
Comment (kaikreuzer) [inline bundles/org.openhab.core.automation/src/main/resources/OH-INF/automation/moduletypes/ItemActions.json:62]:
```suggestion
			"label": "update an item state",
```

File: bundles/org.openhab.core.automation/src/main/resources/OH-INF/automation/moduletypes/ItemActions.json
Comment (kaikreuzer) [inline bundles/org.openhab.core.automation/src/main/resources/OH-INF/automation/moduletypes/ItemActions.json:113]:
```suggestion
					"description": "state that the item will be set to."
```

**Review by kaikreuzer — CHANGES_REQUESTED** [review]

**Comment by cweitkamp** [comment]
Thanks for your quick review. I incorporated the suggested changes.

File: bundles/org.openhab.core.automation/src/main/java/org/openhab/core/automation/internal/module/handler/ItemStateUpdateActionHandler.java
Comment (kaikreuzer) [inline bundles/org.openhab.core.automation/src/main/java/org/openhab/core/automation/internal/module/handler/ItemStateUpdateActionHandler.java:59]: Looking closer into the code, I am not 100% certain that the module configuration cannot be changed during the lifetime of the handler. I at least found `ConfigurationNormalizer.normalizeConfiguration`, which reads the configuration object and modifies it. I'd hence feel safer if this would be re-read in `execute`, just as it is done for the `ItemCommandActionHandler`.

**Review by kaikreuzer — CHANGES_REQUESTED** [review]
Thanks for the quick fixes.

While I think this is indeed a missing feature, I'd still believe it is safer to not merge it last minute and keep it for the next release, especially as I am not sure whether we wouldn't also need a UI adaption to make it appear in the "item" section. Wdyt?

**Comment by ghys** [comment]
I can make a quick UI update this afternoon if this gets merged, it won't take long and shouldn't be dangerous.

**Comment by kaikreuzer** [comment]
I would love to at least have another final final final snapshot build that we all test once more before starting the GA build tonight, so time is really getting tight here.

If you (@openhab/core-maintainers & @openhab/webui-maintainers) agree that we should squeeze it in, I'd still be ok with that, but can also live with postponing it. So feel free to give your go or veto here.

**Comment by wborn** [comment]
I'm with the give it a go people. 👍

**Comment by kaikreuzer** [comment]
@cweitkamp So mainly up to you - if you think it should go in, please address my last comment and ping @ghys to prepare an according UI change.

**Comment by ghys** [comment]
You mean like this 😄 

image

I have some last minute things to merge in that area anyway.

**Comment by ghys** [comment]
The example above results in:

image

File: bundles/org.openhab.core.automation/src/main/java/org/openhab/core/automation/internal/module/handler/ItemStateUpdateActionHandler.java
Comment (cweitkamp) [reply bundles/org.openhab.core.automation/src/main/java/org/openhab/core/automation/internal/module/handler/ItemStateUpdateActionHandler.java]: I applied these suggestions. But I have to investigate later on that because IIRC I contributed other automation modules doing it in the same way.

**Comment by cweitkamp** [comment]
@ghys ping :wink:

> please address my last comment and ping @ghys to prepare an according UI change

Done.

**Review by kaikreuzer — APPROVED** [review]
Thanks @cweitkamp!
I'll merge and trigger the official core build right away now, so we do not loose time waiting for the PR build.

**Comment by kaikreuzer** [comment]
We are not really lucky with the core build - it now went twice into the 2 hour timeout :-(
It does not seem to be blocked by anything, it is just extremely slow...

**Comment by kaikreuzer** [comment]
Cleaned workspace, restarted Jenkins and it succeeded in 47 minutes: https://ci.openhab.org/job/openHAB-Core/629/

Now let's just wait for the final distro: https://ci.openhab.org/job/openHAB3-Distribution/2096/

**Pull request merged**
